### PR TITLE
Upgrade embedded Kotlin version to 1.3.71

### PR DESCRIPTION
**Pull request by bamboo**
https://github.com/JetBrains/kotlin/releases/tag/v1.3.71

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -2265,6 +2265,11 @@
             <sha256 value="eabaa0ba7428117e8c5cd0188c76d2cc8c92e9b331ee0aee83f815fe8f4d76fa"/>
          </artifact>
       </component>
+      <component group="org.gradle.kotlin" name="plugins" version="1.3.5">
```
Comment (eskatos) [inline gradle/verification-metadata.xml]: :x: verification metadata for 1.3.3 shouldn't be needed anymore, let's remove it

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 2]
```diff
@@ -78,7 +78,7 @@ The dependencies for annotation processors are not added to an IDEA module's cla
 
 ==== Updates to bundled Gradle dependencies
 
-- Kotlin has been updated to https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/[Kotlin 1.3.70].
+- Kotlin has been updated to https://github.com/JetBrains/kotlin/releases/tag/v1.3.71[Kotlin 1.3.70].
```
Comment (eskatos) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: :x: Gradle 6.3 was released with 1.3.70. This should stay unchanched. Instead a section `Updates to bundled Gradle dependencies` with the upgrade to 1.3.71 should be added in `Upgrading from 6.3 above.

File: subprojects/kotlin-dsl-plugins/kotlin-dsl-plugins.gradle.kts [hunk 3]
```diff
@@ -149,3 +149,12 @@ tasks.noDaemonIntegTest {
...
 }
+
+// TODO: workaround for https://github.com/gradle/gradlecom/issues/627
+//  which causes `publishPlugins` to fail with:
+//  > java.io.FileNotFoundException: .../subprojects/kotlin-dsl-plugins/src/main/java (No such file or directory)
+afterEvaluate {
+    configurations.archives.get().allArtifacts.removeIf {
+        it.name == "java"
+    }
+}
```
Comment (eskatos) [inline subprojects/kotlin-dsl-plugins/kotlin-dsl-plugins.gradle.kts]: 👍

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (bamboo) [reply gradle/verification-metadata.xml]: Done.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc (hunk 2 above)
Comment (bamboo) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc]: Done. Thanks!

**Comment by bamboo** [comment]
All suggestions applied, @eskatos, thank you very much for the review.

**Pull request merged**